core/custom_field_api.php

```diff
@@ -1113,6 +1113,10 @@ function custom_field_validate( $p_field_id, $p_value ) {
 
 			break;
 		case CUSTOM_FIELD_TYPE_DATE:
+            # Date field can be left blank (when field is not required)
```

Comment:
Why is this check needed?  The following check is valid if value is empty, null or 0.
Also please use tabs for indentation instead of spaces.